app/grandchallenge/components/models.py

```diff
@@ -648,9 +694,24 @@ def create_instance(self, *, image=None, value=None, fileobj=None):
 
     def clean(self):
         super().clean()
+        self._clean_overlay_segments()
         self._clean_store_in_database()
         self._clean_relative_path()
 
+    def _clean_overlay_segments(self):
+        try:
+            ci = ComponentInterface.objects.get(pk=self.pk)
+        except ComponentInterface.DoesNotExist:
+            return
```

Comment:
This is to get the current value for the overlay segments before saving a new value. Of course, this could be anewly created interface, in which case there is no current db value.